R/algorithm.R

```diff
@@ -25,101 +25,101 @@
       logic = "c_spec == 8",
       comments = "`TRUE` when the department where the recorded diagnosis was endocrinology."
     ),
-    lpr2_is_medical_department = list(
+    lpr2_is_medical_dept = list(
       register = "lpr_adm",
       title = "LPR2 other medical department",
       logic = "c_spec %in% c(1:7, 9:30)",
       comments = "`TRUE` when the diagnosis was recorded at a medical department other than endocrinology."
     ),
-    lpr2_has_pregnancy_event = list(
+    lpr2_is_pregnancy_code = list(
       register = "lpr_diag",
       title = "LPR2 diagnoses codes for pregnancy-related outcomes",
       logic = "c_diag =~ '^(DO0[0-6]|DO8[0-4]|DZ3[37])'",
       comments = "These are recorded pregnancy endings like live births and miscarriages."
     ),
-    lpr3_is_endocrinology_department = list(
+    lpr3_is_endocrinology_dept = list(
       register = "kontakter",
       title = "LPR3 endocrinology department",
       # TODO: We will need to make sure the Unicode character gets selected properly in real data.
       logic = "na_if(hovedspeciale_ans, NOT (hovedspeciale_ans %in% c('medicinsk endokrinologi', 'blandet medicin og kirurgi', 'intern medicin', 'geriatri', 'hepatologi', 'h\u00e6matologi', 'infektionsmedicin', 'kardiologi', 'medicinsk allergologi', 'medicinsk gastroenterologi', 'medicinsk lungesygdomme', 'nefrologi', 'reumatologi', 'palliativ medicin', 'akut medicin', 'dermato-venerologi', 'neurologi', 'onkologi', 'fysiurgi', 'tropemedicin'))) == 'medicinsk endokrinologi'",
       comments = "`TRUE` when the department is endocrinology, `FALSE` when it is other medical departments, and missing in all other cases."
     ),
-    lpr3 = list(
+    lpr3_is_needed_code = list(
       register = "diagnoser",
-      title = "Relevant LPR3 diagnoses codes",
+      title = "LPR3 codes used throughout the algorithm",
       logic = "diagnosekode =~ '^(DO0[0-6]|DO8[0-4]|DZ3[37]|DE1[0-4])' AND (diagnosetype == 'A' OR diagnosetype == 'B') AND (senere_afkraeftet == 'Nej')",
       comments = "`A` `diagnosekode` means primary diagnosis and `senere_afkraeftet` means diagnosis was later retracted."
     ),
-    lpr3_has_t1d = list(
+    lpr3_is_t1d_code = list(
       register = "diagnoser",
       title = "LPR3 diagnoses codes for T1D",
       logic = "diagnosekode =~ '^(DE10)'",
       comments = ""
     ),
-    lpr3_has_t2d = list(
+    lpr3_is_t2d_code = list(
       register = "diagnoser",
       title = "LPR3 diagnoses codes for T2D",
       logic = "diagnosekode =~ '^(DE11)'",
       comments = ""
     ),
-    no_pregnancy = list(
+    is_not_within_pregnancy_period = list(
       register = NA,
-      title = "Remove events within a potential pregnancy period",
+      title = "Events that are not within a potential pregnancy period",
       logic = "NOT (is_pregnancy_code AND has_elevated_hba1c AND (date >= (pregnancy_event_date - weeks(40)) OR date <= (pregnancy_event_date + weeks(12)))",
```

Comment:
This indicates that the logic for pregnancy periods is only applied to elevated HbA1c values. It should also be applied to GLD purchases. Would something like this work?
```suggestion
      logic = "NOT (is_pregnancy_code AND (has_elevated_hba1c OR has_gld_purchases) AND (date >= (pregnancy_event_date - weeks(40)) OR date <= (pregnancy_event_date + weeks(12)))",
```

Reply:
Good point. I'll make a separate issue for this, since it's beyond the scope of this PR.